pkg/panels/kubernetes/globals.go

```diff
@@ -2,94 +2,107 @@ package kubernetes
+	return "job=\"" + NODE_EXPORTER_LABEL_VALUE + "\""
 }
 
 // GetControllerManagerMatcher returns the matcher for the controller-manager job.
 func GetControllerManagerMatcher() string {
-	return CONTROLLER_MANAGER_MATCHER
+	return "job=\"" + CONTROLLER_MANAGER_LABEL_VALUE + "\""
 }
 
 // GetSchedulerMatcher returns the matcher for the scheduler job.
 func GetSchedulerMatcher() string {
-	return KUBE_SCHEDULER_MATCHER
+	return "job=\"" + KUBE_SCHEDULER_LABEL_VALUE + "\""
 }
 
 // GetKubeProxyMatcher returns the matcher for the kube-proxy job.
 func GetKubeProxyMatcher() string {
-	return KUBE_PROXY_MATCHER
+	return "job=\"" + KUBE_PROXY_LABEL_VALUE + "\""
 }
 
-// SetCAdvisorMatcher sets the matcher for the cadvisor job globally.
-func SetCAdvisorMatcher(matcher string) {
-	CADVISOR_MATCHER = matcher
+// NOTE: Matcher setter functions have been removed since matchers are now computed dynamically.
+// Use the SetXXXLabelValue functions instead to change the underlying label values.
+
+// SetAPIServerLabelValue sets the label value for the api server job globally.
+// WARNING: Ensure you only set this to value that represents a Kube API Server specifically.
+func SetAPIServerLabelValue(labelValue string) {
+	API_SERVER_LABEL_VALUE = labelValue
 }
 
-// SetKubeStateMetricsMatcher sets the matcher for the kube-state-metrics job globally.
-func SetKubeStateMetricsMatcher(matcher string) {
-	KUBE_STATE_METRICS_MATCHER = matcher
+// SetKubeletLabelValue sets the label value for the kubelet job globally.
+// WARNING: Ensure you only set this to value that represents a Kubelet specifically.
+func SetKubeletLabelValue(labelValue string) {
+	KUBELET_LABEL_VALUE = labelValue
 }
 
-// SetKubeletMatcher sets the matcher for the kubelet job globally.
-func SetKubeletMatcher(matcher string) {
-	KUBELET_MATCHER = matcher
+// SetNodeExporterLabelValue sets the label value for the node-exporter job globally.
+// WARNING: Ensure you only set this to value that represents a Node Exporter specifically.
+func SetNodeExporterLabelValue(labelValue string) {
+	NODE_EXPORTER_LABEL_VALUE = labelValue
 }
 
-// SetNodeExporterMatcher sets the matcher for the node-exporter job globally.
-func SetNodeExporterMatcher(matcher string) {
-	NODE_EXPORTER_MATCHER = matcher
+// SetControllerManagerLabelValue sets the label value for the controller-manager job globally.
+// WARNING: Ensure you only set this to value that represents a Controller Manager specifically.
+func SetControllerManagerLabelValue(labelValue string) {
+	CONTROLLER_MANAGER_LABEL_VALUE = labelValue
 }
 
-// SetControllerManagerMatcher sets the matcher for the controller-manager job globally.
-func SetControllerManagerMatcher(matcher string) {
-	CONTROLLER_MANAGER_MATCHER = matcher
+// SetCadvisorLabelValue sets the label value for the cadvisor job globally.
+// WARNING: Ensure you only set this to value that represents a Cadvisor specifically.
+func SetCadvisorLabelValue(labelValue string) {
```

Comment:
Nitpick: different capitalization in `GetCAdvisorMatcher` vs `SetCadvisorLabelValue`: let's stick to one version